src/tests/integration_tests/test_entity_citation_validator.py

```diff
@@ -0,0 +1,99 @@
+from unittest.mock import patch
+
+import pytest
+from loguru import logger
+from nltk.metrics import jaccard_distance
+
+import sherpa_ai.config as cfg
+from sherpa_ai.agents.qa_agent import QAAgent
+from sherpa_ai.events import EventType
+from sherpa_ai.memory import SharedMemory
+from sherpa_ai.models.sherpa_base_chat_model import SherpaChatOpenAI
+from sherpa_ai.output_parsers.entity_validation import EntityValidation
+
+# from sherpa_ai.test_utils.llms import get_llm
+from sherpa_ai.tools import SearchTool
+from sherpa_ai.utils import extract_entities
+
+
+@pytest.mark.parametrize(
+    "objective , input_data, expected_entities",
+    [
+        (
+            "what is unique about Ethiopia calendar?  based on USA calendar assosation. ",
+            (
+                """
+                According to Ethiopian calendar Assosation (ECA) , Ethiopia has thirteen months. Canada and Kenya also mentioned this. also recognized by World Calendar Assosation (WCA) """,
+                [
+                    {
+                        "Document": "Ethiopian calendar",
+                        "Source": "https://www.sabioholding.com/press-releases/sabio-delivers-11-q2-2023-revenue-growth-led-by-57-increase-in-connected-tv-ott-sales",
+                    }
+                ],
+            ),
+            ["Ethiopian", "Ethiopian Calendar Association", "kenya"],
+        ),
+        (
+            "Tell me a fact about Star Trek: The Next Generation?",
+            (
+                """Fact: In Star Trek: The Next Generation (STNG), the United Federation of Planets, a coalition of various planetary governments working for peace and cooperation, establishes a set of interstellar laws known as the "Federation Charter." This document outlines the principles and regulations governing member worlds and their interactions. """,
+                [
+                    {
+                        "Document": "Star Trek: The Next Generation",
+                        "Source": "https://www.starTrek.com",
+                    }
+                ],
+            ),
+            ["STNG", "Star Trek", "the United Federation of Planets"],
+        ),
+    ],
+)
+def test_entity_citation_succeeds_in_qa(input_data, expected_entities, objective):
+    # noqa: F811
+    # llm = get_llm(__file__, test_number_citation_succeeds_in_qa.__name__)
```

Comment:
use get_llm

Reply:
I will but the whole experiment of this is to try to make the llm to correct itself. so if I use the get_llm won't it just make it not fail at all.